ReactSkia/core_modules/RSkKeyboardObserver.cpp

```diff
@@ -0,0 +1,61 @@
+#include "RSkKeyboardObserver.h"
+#include "ReactSkia/sdk/NotificationCenter.h"
+namespace facebook {
+namespace react {
+using namespace std;
+RSkKeyboardObserver::RSkKeyboardObserver(
+      const std::string &name,
+      std::shared_ptr<CallInvoker> jsInvoker,
+      Instance *bridgeInstance)
+      : RSkEventEmitter(name, jsInvoker, bridgeInstance){
+      }
+
+void RSkKeyboardObserver::handleKeyboardEventNotification(KeyboardEventType eventType){
+  RNS_LOG_DEBUG("[handleKeyboardEventNotification]");
+  sendKeyboardEventWithName(eventType);
+}
+
+void RSkKeyboardObserver::startObserving(){
+  RNS_LOG_DEBUG("[startObserving]");
+    //Add listerners
+  std::function<void (KeyboardEventType)> keyboardNotification  = std::bind(&RSkKeyboardObserver::handleKeyboardEventNotification, this,std::placeholders::_1);
+  keyboardEventId_ = NotificationCenter::subWindowCenter().addListener("KeyboardEventNotification", keyboardNotification);
+
+}
+
+void RSkKeyboardObserver::stopObserving(){
+  RNS_LOG_DEBUG("[stopObserving]");
+  NotificationCenter::subWindowCenter().removeListener(keyboardEventId_);
+}
+
+void RSkKeyboardObserver::sendKeyboardEventWithName(KeyboardEventType eventType){
+  string KeyboardEventType;
+  switch(eventType){
+    case keyboardWillShow:
+      KeyboardEventType = string("keyboardWillShow");
```

Comment:
cant we directly send the event string to the notification center from OSK module.this will avoid this switch case